src/serde/mod.rs

```diff
@@ -44,9 +44,10 @@ pub fn from_robtop_str<'a, T: HasRobtopFormat<'a>>(input: &'a str) -> Result<T,
 }
 
 pub fn to_robtop_data<'a, T: HasRobtopFormat<'a>>(t: &'a T) -> Result<Vec<u8>, SerError> {
```

Comment:
If I add the method for the String I'd prefer using the checked method, it's pretty much the only unsafe we use. Otherwise we let the user decide whether to convert it to a String

Reply:
Yeah, the default should definitely use the checked conversation, if the user wants to drop down to unsafe they have to manually do so by writing to a `Vec<u8>` and calling `from_utf8_unchecked` themselves